lib/bitcoin/lock/LockMonitor.ts

```diff
@@ -51,8 +58,30 @@ export default class LockMonitor {
     await this.periodicPoll();
   }
 
-  private async periodicPoll (): Promise<void> {
+  /**
+   * Gets the current lock information if exist; undefined otherwise. Throws an error
+   * if the lock information is not confirmed on the blockchain.
+   */
+  public getCurrentValueTimeLock (): ValueTimeLockModel | undefined {
+
+    // Make a copy of the state so in case it gets changed between now and the function return
+    const currentLockState = Object.assign({}, this.currentLockState!);
+
+    // If there's no lock then return undefined
+    if (currentLockState.status === LockStatus.None) {
+      return undefined;

```

Comment:
Would it make sense to throw an error here instead of undefined like line 78?

Reply:
You are right ... throwing it upon refactor.